docs/guides/server/tracing.adoc

```diff
@@ -0,0 +1,105 @@
+It means the format of the log records is changed and may start like follows:
+
+[source, bash]
+----
+2024-08-05 15:27:07,144 traceId=b636ac4c665ceb901f7fdc3fc7e80154, parentId=d59cea113d0c2549, spanId=d59cea113d0c2549, sampled=true WARN  [org.keycloak.events] ...
+----
+
+=== Hide trace info in logs
+
+You can hide tracing information in specific log handlers by specifying their associated {project_name} option `log-<handler-name>-include-trace`, where `<handler-name>` is the name of the log handler.
+For instance, to disable trace info in the `console` log, you can turn it off as follows:
+
+<@kc.start parameters="--tracing-enabled=true --log=console --log-console-include-trace=false"/>
+
+NOTE: When you explicitly override the log format for the particular log handlers, the `*-include-trace` options do not have any effect, and no tracing is printed out.
+
+== Sampling
+
+Sampler decides whether a trace should be discarded or forwarded, effectively managing noise and reducing overhead by limiting the number of collected traces sent to the collector.
+It helps manage resource consumption, which leads to avoiding the huge storage costs of tracing every single request and potential performance penalty.
+
+WARNING: For a production-ready environment, sampling should be properly set to minimize infrastructure costs.
+
+{project_name} supports several built-in OpenTelemetry samplers, such as:
+
+<@opts.expectedValues option="tracing-sampler-type"/>
+
+The used sampler can be changed via the `tracing-sampler-type` property.
+
+=== Default sampler
+The default sampler for {project_name} is `traceidratio`, which controls the rate of trace sampling based on a specified ratio configurable via the `tracing-sampler-ratio` property.
+
+==== Trace ratio
+The default trace ratio is `1.0`, which means all traces are sampled - sent to the collector.
+The ratio is a floating number in the range `(0,1]`.
+For instance, when the ratio is `0.1`, only 10% of the traces are sampled.
+
+WARNING: For a production-ready environment, the trace ratio should be a smaller number to prevent the massive cost of trace store infrastructure and avoid performance overhead.
+
+==== Rationale
+
+The sampler makes its own sampling decisions based on the current ratio of sampled spans, regardless of the decision made on the parent span, as with using the `parentbased_traceidratio` sampler.
+The `parentbased_traceidratio` sampler could be the preferred default type, but it would require additional aspects to ensure safety.
+
+There might be potential security risks related to information exposure or DoS attacks - proper HTTP filtering and adequate measures of caller trust would need to be assessed.
+For more information, see the https://www.w3.org/TR/trace-context/#security-considerations[W3C Trace context] document.
```

Comment:
IMHO this is a bit vague. Can you be a more specific here describing both the benefits and the risks of the `parentbased_traceidratio`?